src/modules/hip/hip_advanced_augmentation.cpp

```diff
@@ -0,0 +1,924 @@
+
+    return RPP_SUCCESS;
+}
+
+RppStatus
+non_linear_blend_hip_batch_tensor(Rpp8u *srcPtr1, Rpp8u *srcPtr2, Rpp8u *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info)
+{
+    int in_plnpkdind = getplnpkdind(tensor_info._in_format), out_plnpkdind = getplnpkdind(tensor_info._out_format);
+    InitHandle *handle_obj = handle.GetInitHandle();
+    Rpp32u max_height, max_width;
+    max_size(handle_obj->mem.mgpu.csrcSize.height, handle_obj->mem.mgpu.csrcSize.width, handle.GetBatchSize(), &max_height, &max_width);
+
+#if defined (HIPRTC)
+
+    std::vector<size_t> vld{16, 16, 1};
+    std::vector<size_t> vgd{max_width, max_height, handle.GetBatchSize()};
+    std::string kernel_file = "non_linear_blend.cpp";
+    std::string kernel_name = "non_linear_blend_batch";
+    get_kernel_name(kernel_name, tensor_info);
+
+    handle.AddKernel("", "", kernel_file, kernel_name, vld, vgd, "")(srcPtr1,
+                                                                     srcPtr2,
+                                                                     dstPtr,
+                                                                     handle_obj->mem.mgpu.floatArr[0].floatmem,
+                                                                     handle_obj->mem.mgpu.roiPoints.x,
+                                                                     handle_obj->mem.mgpu.roiPoints.roiWidth,
+                                                                     handle_obj->mem.mgpu.roiPoints.y,
+                                                                     handle_obj->mem.mgpu.roiPoints.roiHeight,
+                                                                     handle_obj->mem.mgpu.srcSize.height,
+                                                                     handle_obj->mem.mgpu.srcSize.width,
+                                                                     handle_obj->mem.mgpu.maxSrcSize.width,
+                                                                     handle_obj->mem.mgpu.srcBatchIndex,
+                                                                     tensor_info._in_channels,
+                                                                     handle_obj->mem.mgpu.inc,
+                                                                     handle_obj->mem.mgpu.dstInc,
+                                                                     in_plnpkdind,
+                                                                     out_plnpkdind);
+
+#elif defined(STATIC)
+    hip_exec_non_linear_blend_batch(srcPtr1, srcPtr2, dstPtr, handle, tensor_info, in_plnpkdind, out_plnpkdind, max_height, max_width);
+#endif
+    return RPP_SUCCESS;
+}
+
+RppStatus
+non_linear_blend_hip_batch_tensor_fp16(Rpp16f *srcPtr1, Rpp16f *srcPtr2, Rpp16f *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info)
+{
+//     int in_plnpkdind = getplnpkdind(tensor_info._in_format), out_plnpkdind = getplnpkdind(tensor_info._out_format);
```

Comment:
For all the fp16 functions (7-1=6 funcs), keep the contents of the function uncommented here in the higher level API. Let's just comment out the contents of the kernel itself.

Reply:
done